imgui.lua

```diff
@@ -45,8 +45,13 @@ end
 imgui.Hook("PreRender", "Input", function()
 	-- calculate mouse state
 	if shouldAcceptInput() then
+		local useBind = input.LookupBinding("use")
+		local attackBind = input.LookupBinding("attack")
```

Comment:
Final nitpick but I wonder if passing `true` as the second argument and prefixing these with + would optimize the calls tiny bit further? Exact matching sounds like it could be faster without requiring fuzzy matching logic

https://wiki.facepunch.com/gmod/input.LookupBinding

Reply:
Can't find exact matching logic in C++ logic, so maybe it's some Garry addition: https://github.com/ValveSoftware/source-sdk-2013/blob/0d8dceea4310fde5706b3ce1c70609d72a38efdf/mp/src/vgui2/vgui_controls/Panel.cpp#L2166

Binding finding seems to be O(n) :/ Oh well, it's still a fast operation.

Reply:
Considering a single call takes ~3ns I don't think it can get much faster, but I'll do it anyways to grind out another nanosecond hopefully ;)

Reply:
Looks like there will still be an issue with people who bind multiple keys to the same binding. This is much better than before at least.

Reply:
Ah, that's true. I'll try to find some time to see if I can reproduce them. If you'd like to try tracking them down, then trying a) using imgui while driving a car, b) using imgui while the context menu is open, c) using imgui while some other panel (e.g. chat) is open would be some of the situations to check. Although I don't think the chat case should necessarily even work

Reply:
I tested the cases you listed, and it does seem that when in a vehicle imgui works with either method.
Other than that I haven't seen a problem with any other panels so far or the context menu considering `ShouldAcceptInput` blocks the input in those cases, as long as the panel is making use of the `MakePopup` method.

For now in my own version I'll continue using KeyDown and see if I can find any issues.

Reply:
I found the commit in preceding library where KeyDown was changed: https://github.com/wyozi/3d2d-imgui/commit/0bb4a9a28aabd0a0d8bddd4e0c95e90da4074da4 but no explanation why.

If KeyDown works fine in prod for a day or two, feel free to make a PR. If issues pop up they're easier to document and fix at that time then

Reply:
At least for mouse presses (+attack) input.IsKeyDown would make sense if they should still work with gui.EnableScreenClicker

Reply:
That's some old code, correct me if I'm wrong but it looks like how it used to be the current `ShouldAcceptInput` method is implemented in a different way. Seems to me like that itself may have fixed many issues you would have had with it in the past.

I tested both methods with `gui.EnableScreenClicker` and it seems that `ShouldAcceptInput` blocks the inputs for this, so it's not an issue with `Player:KeyDown`.

I haven't had any issues so far with `KeyDown` otherwise. I say it'd be worth it to use it and fix any new issues that come up because of it.

Reply:
I'll wait another day or so before I create a new pull request for `KeyDown` in case any issues pop up or if you remember anything else to test.